MAVProxy/modules/mavproxy_DGPS.py

```diff
@@ -7,39 +7,75 @@
 from pymavlink import mavutil
 from MAVProxy.modules.lib import mp_module
 
+import socket, errno
+from pymavlink import mavutil
+from MAVProxy.modules.lib import mp_module
+
 class DGPSModule(mp_module.MPModule):
     def __init__(self, mpstate):
-        super(DGPSModule, self).__init__(mpstate, "DGPS", "DGPS injection support")
+        super(DGPSModule, self).__init__(mpstate, "DGPS", "DGPS injection support for SBP/RTCP/UBC")
         self.portnum = 13320
         self.port = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
         self.port.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
         self.port.bind(("127.0.0.1", self.portnum))
         mavutil.set_close_on_exec(self.port.fileno())
         self.port.setblocking(0)
-        print "Listening for DGPS packets on UDP://%s:%s" % ("127.0.0.1", self.portnum)
+        self.inject_seq_nr = 0
+        print "DGPS: Listening for RTCM packets on UDP://%s:%s" % ("127.0.0.1", self.portnum)
+    
+    def send_rtcm_msg(self, data):
+        msglen = 180;
+
+        if (len(data) > msglen * 4):
+            print "DGPS: Message too large", len(data)
+            return
+
+        msgs = (len(data) / msglen) + 1
+        if (len(data) % msglen == 0):
+            msgs = len(data) / msglen
+
+        for a in range(0, msgs):
+
+            flags = 0
+
+            if (msgs) > 0:
+                flags = 1
+
+            flags |= (a & 0x3) << 1
```

Comment:
Probably worth adding a comment that says this is setting the fragment ID, the one below is sequence and the one above is the is fragmented flag.

If the packet isn't fragmented you can opt to not set any of the bits in the packet, which might be worthwhile, I'm not sure if PX4 looks at the fragment bit or not (and if they look at the bit vs looking at the whole flags field)